### PR TITLE
ON-13368 # Added cancelSchedulingBooking fn

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: src/services/api/scheduling.ts [hunk 1]
```diff
@@ -55,19 +59,57 @@ function generateSchedulingConfiguration({
       error,
     )
     switch (error.status) {
-      case 401: {
```
Comment (mymattcarroll) [inline src/services/api/scheduling.ts:58]: 401 case has been removed?

Comment (Zaxist) [reply src/services/api/scheduling.ts:58]: They are unathenticated routes

File: src/scheduling-service.ts [hunk 2]
```diff
@@ -150,9 +155,18 @@ async function handleSchedulingSubmissionEvent({
   return scheduling
 }
 
+async function handleCancelSchedulingBooking(details: {
+  submissionId: string
+  nylasEditHash: string
+  reason: string
+}) {
+  return await cancelSchedulingBooking(details)
+}
```
Comment (mymattcarroll) [inline src/scheduling-service.ts:164]: Remove this?

**Comment by mymattcarroll** [comment]
Need docs for new function.

File: src/services/api/scheduling.ts (hunk 1 above)
Comment (mymattcarroll) [reply src/services/api/scheduling.ts:58]: `/scheduling/generate-booking-url` will return a 401 if the form requires auth and the user has not logged in and they somehow run this function.

Comment (Zaxist) [reply src/services/api/scheduling.ts:58]: ah

File: src/submission-service.ts [hunk 3]
```diff
@@ -128,11 +128,14 @@ async function processPendingQueue(): Promise<void> {
 async function submit({
   formSubmission,
   paymentReceiptUrl,
-  schedulingReceiptUrl,
+  schedulingUrlConfiguration,
```
Comment (mymattcarroll) [inline src/submission-service.ts:131]: Docs this need to change as well.

Comment (Zaxist) [reply src/submission-service.ts:131]: This is currently not in any docs... I suppose I'll add them

Comment (Zaxist) [reply src/submission-service.ts:131]: nor are half the functions in the scheduling service

Comment (mymattcarroll) [reply src/submission-service.ts:131]: I only documented `handleSchedulingQuerystring()` as the other functions in the service should not be used externally.

Comment (mymattcarroll) [reply src/submission-service.ts:131]: We could move them to another file, if we don't want them exported on the `schedulingService`

Comment (Zaxist) [reply src/submission-service.ts:131]: Yeh we should do that, if they are not meant for use

**Pull request merged**